### ISSUE TITLE
FP: in multiple cases the result can be evaluated as false, consider hardening the condition

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#1707]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| **Plugin**         | U3.0.11.1 |
| **Language level** | PHP 7.4                 |

```php
// [EA] In multiple cases the result can be evaluated as false, consider hardening the condition.
$nullableVariableOfSomeKind = null; // eg. int|null

return $nullableVariableOfSomeKind < 0;
```

The inspection says that in multiple cases the result can be evaluated as false. Which cases, for instance? Actually, the major issue here is when we use `<=` or `>=` because it could be `true` for `null` values.

I have a real case like that:

```php
$intNullable = 123; 

// [EA] In multiple cases the result can be evaluated as false, consider hardening the condition.
if ($intNullable !== null && $intNullable < 0) {}
```

**Comment by rentalhost** [comment]
Hm, I guess I understand some possible cases:

```php
var_dump( strpos('abc', 'd') < 1000 ); 
```

So I think that this inspection could be ignored if value is `0`, because it is the unique value that could be confused with `false`,

**Comment by rentalhost** [comment]
Meanwhile, some other false-positives should be fixed automatically when Jetbrains fixes that: https://youtrack.jetbrains.com/issue/WI-59346.

**Comment by donquixote** [comment]
I have another scenario here.
It seems a very specific edge case, as if "<" and ">" are treated differently with object properties.

```php
<?php

declare(strict_types=1);

class C {

  private ?int $min;

  private ?int $max;

  public function __construct(?int $min, ?int $max) {
    $this->min = $min;
    $this->max = $max;
  }

  public function check(int $number, ?int $min, ?int $max): bool {

    // No inspection warning.
    if ($this->min !== NULL && $number < $this->min) return FALSE;

    // Inspection warning: "[EA] In multiple cases the result can be evaluated as false, consider hardening the condition."
    if ($this->max !== NULL && $number > $this->max) return FALSE;

    // No inspection warning.
    if ($min !== NULL && $number < $min) return FALSE;

    // No inspection warning.
    if ($max !== NULL && $number > $max) return FALSE;

    return TRUE;
  }
}

```

> PhpStorm 2021.2.1
> Build #PS-212.5080.71, built on August 26, 2021
> Runtime version: 11.0.11+9-b1504.16 amd64
> VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
> Linux 5.11.0-31-generic
> GC: ParNew, ConcurrentMarkSweep
> Memory: 3903M
> Cores: 8
> Registry: run.processes.with.pty=TRUE, search.everywhere.settings=true, ide.tooltip.initialDelay=952
> Non-Bundled Plugins: de.espend.idea.php.annotation (8.0.0), com.kalessil.phpStorm.phpInspectionsUltimate-mp (2021.2)
> Current Desktop: KDE

> PHP Inspections (EA Ultimate) 2021.2
